pkgs/by-name/le/lexmark-aex/package.nix

```diff
@@ -0,0 +1,76 @@
+{ lib
+, stdenv
+, cups
+, fetchurl
+, patchPpdFilesHook
+, autoPatchelfHook
+, perl
+, avahi
+}:
+
+stdenv.mkDerivation {
+  pname = "lexmark-aex";
+  version = "1.0";
+
+  dontPatchELF = true;
+  dontStrip = true;
+
+  src = fetchurl {
+    url = "https://downloads.lexmark.com/downloads/drivers/Lexmark-AEX-PPD-Files.tar.Z";
+    hash = "sha256-jfBORC5M+ep/Q1W25xT4koBo6WTpmjeRvK1yUd6DLro=";
+  };
+
+  strictDeps = true;
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    patchPpdFilesHook
+  ];
+
+  buildInputs = [
+    # Needed for autoPatchelfHook.
+    avahi
+    cups
+    # Needed for patchShebangs.
+    perl
+  ];
+
+  # Needed for autoPatchelfHook.
+  runtimeDependencies = [ (lib.getLib cups) ];
+
+  ppdFileCommands = [ "CommandFileFilterG2" "rerouteprintoption" ];
+
+  installPhase = let
+    libdir =
+      if stdenv.system == "x86_64-linux"    then "lib64"
+      else if stdenv.system == "i686_linux" then "lib"
+      else throw "other platforms than i686_linux and x86_64-linux are not yet supported";
+  in ''
+    runHook preInstall
+    cd GlobalPPD_1.4
+
+    # Install raster image filter.
+    install -Dm755 rerouteprintoption $out/lib/cups/filter/rerouteprintoption
+    patchShebangs $out/lib/cups/filter/rerouteprintoption
+
+    # Install additional binary filters.
+    for i in CommandFileFilterG2 LexHBPFilter; do
+      install -Dm755 "${libdir}/$i" "$out/lib/cups/filter/$i"
+    done
+
+    # Install PPD.
+    install -Dm 0644 -t $out/share/cups/model/Lexmark *.ppd
+
+    cd ..
+    runHook postInstall
```

Comment:
cd-ing into a dir is kinda meh and usually the preferred thing is to just write the full paths or to set a env var and prefix the paths with it